sdk/core/Azure.Core/src/Shared/DiagnosticScopeFactory.cs

```diff
@@ -16,11 +16,13 @@ internal class DiagnosticScopeFactory
         private static Dictionary<string, DiagnosticListener>? _listeners;
         private readonly string? _resourceProviderNamespace;
         private readonly DiagnosticListener? _source;
+        private readonly bool _suppressNestedClientScopes;
 
-        public DiagnosticScopeFactory(string clientNamespace, string? resourceProviderNamespace, bool isActivityEnabled)
+        public DiagnosticScopeFactory(string clientNamespace, string? resourceProviderNamespace, bool isActivityEnabled, bool suppressNestedClientScopes)
```

Comment:
Should we assign a default value here so that new clients fall into the pit of success?